### PR TITLE
Re-enable code coverage report on PR pipeline

### PR DESCRIPTION
**Pull request by asvrada**
Previously with the help of a Jenkins plugin, we can display the code coverage result as a status/checker on the Pull Request.

Now that the plugin has been deprecated, this PR modifies the pipeline to upload the code coverage result to Jenkins and provides documentation on accessing the new report.

Note the XML file generated by Python package `lcov-cobertura` is not used now
Signed-off-by: Zijie Wu <zijiewu@microsoft.com>

### CODE REVIEW

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
try bors

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try-

**Comment by CyanDevs** [comment]
bors try-

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try-

**Comment by asvrada** [comment]
bors try

**Review by radhikaj — APPROVED** [review]
Thanks @asvrada

File: .jenkins/library/vars/tests.groovy
Comment (mingweishih) [inline .jenkins/library/vars/tests.groovy:18]: @CyanDevs Is the agreement that we setup up these dependencies on the fly instead of updating the image?

File: scripts/ansible/roles/linux/openenclave/vars/ubuntu/main.yml
Comment (mingweishih) [inline scripts/ansible/roles/linux/openenclave/vars/ubuntu/main.yml:27]: Shall we also install the `lcov-cobertura` pip package?
As a reference this is the PR I had before: https://github.com/openenclave/openenclave/pull/3138/files

File: .jenkins/library/vars/tests.groovy
Comment (asvrada) [reply .jenkins/library/vars/tests.groovy:18]: We should install these dependencies by updating the image.

**Review by mingweishih — APPROVED** [review]
LGTM

File: scripts/ansible/roles/linux/openenclave/vars/ubuntu/main.yml
Comment (CyanDevs) [reply scripts/ansible/roles/linux/openenclave/vars/ubuntu/main.yml:27]: Are there cases where users would ever need lcov and lcov-cobertura? This Ansible role is used as part of our contributors setup as well and any contributors using those Ansible roles for setup would also install the package.

If the lcov packages are undesirable for contributors I would suggest putting it into the role that is specific to our images: https://github.com/openenclave/openenclave/blob/master/scripts/ansible/roles/linux/jenkins/tasks/packer.yml

File: scripts/ansible/roles/linux/openenclave/tasks/ubuntu/packages-install.yml
Comment (CyanDevs) [inline scripts/ansible/roles/linux/openenclave/tasks/ubuntu/packages-install.yml:32]: It would be preferrable to use the fully qualified name `ansible.builtin.pip`

**Comment by CyanDevs** [comment]
Great visual @asvrada !

**Comment by asvrada** [comment]
> Are there cases where users would ever need lcov and lcov-cobertura? This Ansible role is used as part of our contributors setup as well and any contributors using those Ansible roles for setup would also install the package.

> If the lcov packages are undesirable for contributors I would suggest putting it into the role that is specific to our images: https://github.com/openenclave/openenclave/blob/master/scripts/ansible/roles/linux/jenkins/tasks/packer.yml

Hi @CyanDevs  could you elaborate? Not sure if there is a specific role for code coverage

**Comment by CyanDevs** [comment]
> > Are there cases where users would ever need lcov and lcov-cobertura? This Ansible role is used as part of our contributors setup as well and any contributors using those Ansible roles for setup would also install the package.
> 
> > If the lcov packages are undesirable for contributors I would suggest putting it into the role that is specific to our images: https://github.com/openenclave/openenclave/blob/master/scripts/ansible/roles/linux/jenkins/tasks/packer.yml
> 
> Hi @CyanDevs could you elaborate? Not sure if there is a specific role for code coverage

The Ansible role you are changing will affect any contributors that use Ansible to set up OE (i.e. https://github.com/openenclave/openenclave/blob/master/docs/GettingStartedDocs/Contributors/SGX1FLCGettingStarted.md). My question is whether those contributors would need to run code coverage on their own VM, and if not, then the packages should be installed as part of the packer tasks linked earlier would be better as those tasks are only ran for CI images.

**Comment by asvrada** [comment]
Thanks for clarifying. Will move these package to CI images

**Review by CyanDevs — APPROVED** [review]
LGTM -- please wait for new images to be built before merging.

**Comment by asvrada** [comment]
Sounds good!

> Could you squash the commits?
> 
> Afterwards, please wait for new images to be built before merging. Will let you know when that happens.

Sounds good.

**Comment by CyanDevs** [comment]
This is ready to merge after the merge conflict is resolved.

**Comment by asvrada** [comment]
bors +

**Pull request merged**
